app/src/main/java/com/quran/labs/androidquran/util/ShareUtil.java

```diff
@@ -53,29 +53,51 @@ public void shareViaIntent(Activity activity, String text, @StringRes int titleR
     activity.startActivity(Intent.createChooser(intent, activity.getString(titleResId)));
   }
 
-  public String getShareText(Context context,
+  public String getShareTextperTranslation(Context context,
                                     QuranAyahInfo ayahInfo,
-                                    String[] translationNames) {
+                                    String[] translationNames,String selectedTranslation) {
     final StringBuilder sb = new StringBuilder();
-    if (ayahInfo.arabicText != null) {
-      sb.append(ayahInfo.arabicText)
-        .append("\n\n");
-    }
 
     for (int i = 0, size = ayahInfo.texts.size(); i < size; i++) {
+      if(selectedTranslation ==translationNames[i])
+      {
       if (i < translationNames.length) {
         sb.append('(')
           .append(translationNames[i])
           .append(")\n");
       }
       sb.append(ayahInfo.texts.get(i))
         .append("\n\n");
-    }
+    }}
     sb.append('-')
       .append(quranInfo.getSuraAyahString(context, ayahInfo.sura, ayahInfo.ayah));
 
     return sb.toString();
   }
+  public String getShareText(Context context,
```

Comment:
why do you need this, shouldn't there already be a method for doing this?

Reply:
i firstly combined the two methods in one, but then the logic was not much readable because pertranslation method does not include arabic text,(need an if,and also a loop if for iterate selected translation name in available translation names)and there is a problem with ayahinfo object that it does not include the translation name or db name in text[] object, so i needed to send the translationname as parameter and the total available translations.. the first sharetext method is used for copy alltranslations,so does not need the selected translationname but pertranslation method is used for copy only selected method that needs translationname . so i decided to add the pertranslation method in another method not to mess the method with if expressions. 
anyway  if you prefer i can combine two of them in one?